src/details/ArborX_Ray.hpp

```diff
@@ -198,6 +199,216 @@ bool intersects(Ray const &ray, Box const &box)
   return intersection(ray, box, tmin, tmax) && (tmax >= 0.f);
 }
 
+// rotate to the x-axis
+KOKKOS_INLINE_FUNCTION Point rotate2D(Point const &point)
```

Comment:
The function rotates a 3-D plane which the ray and triangles are coplanar to a 2-D plane, so that the below algorithm can be used to find the intersection points.
https://en.wikipedia.org/wiki/Line%E2%80%93line_intersection#Given_two_points_on_each_line_segment
There might be other better/standard ways to do it.